metadata-ingestion/src/datahub/ingestion/source/ldap.py

```diff
@@ -207,8 +207,10 @@ def handle_user(self, dn: str, attrs: Dict[str, Any]) -> Iterable[MetadataWorkUn
                     self.config.filter,
                     serverctrls=[self.lc],
                 )
-                _m_dn, m_attrs = self.ldap_client.result3(manager_msgid)[1][0]
-                manager_ldap = guess_person_ldap(m_attrs)
+                result = self.ldap_client.result3(manager_msgid)
+                if result[1]:
```

Comment:
wouldn't this also result in an out of range error if the list is empty? since we want to access index 1, we could check `if result.length > 1:`

Reply:
The result is a Tuple and here is an example where the existing code is going to fail (directly from my debug):

(101, [], 3, [<ldap.controls.libldap.SimplePagedResultsControl object …>])

But checking number elelments in the tuple as:  len(result) > 1 will not work because there are 4 elements at this  tuple, and an actual issue is a second [0], when we try to access the first element from [].

Reply:
ahhh gotcha, yeah that was my misunderstanding on what the actual output of `self.ldap_client.result3(manager_msgid)` was. Thanks for clarifying!

Reply:
@chriscollins3456 Sure, glad if this was helpful.